psydac/utilities/utils.py

```diff
@@ -5,13 +5,32 @@
 import numpy as np
 
 #===============================================================================
-def refine_array_1d( x, n ):
+def refine_array_1d(x, n, remove_duplicates=True):
+    """Refines a 1D array by create a subdivision between each point
 
-    xr = []
-    for (a,b) in zip(x[:-1],x[1:]):
-        xr.extend( np.linspace( a, b, n, endpoint=False ) )
-    xr.append( x[-1] )
+    Parameters
+    ----------
+    x : ndarray
+        1D array to be refined
+    n : int
+        The number of points in each subdivision
+
+    remove_duplicates : bool, default=True
+        If True, the subdivision doesn't include the right boundary of the interval
```

Comment:
```suggestion
        If True, the refined array will not contain any duplicate points.
        If False, the original internal grid points x[1:-1] will appear twice: this may
           be useful to visualize fields that are discontinuous across cell boundaries.
```

Reply:
Done in 8301506